bootstrapper/internal/initserver/initserver.go

```diff
@@ -198,7 +203,9 @@ func (s *Server) Init(ctx context.Context, req *initproto.InitRequest) (*initpro
 func (s *Server) Stop() {
 	s.log.Infof("Stopping")
 
+	s.shutdownLock.Lock()
```

Comment:
Could we have a comment here giving some information on why we need to do the locking? This seems like a perfect candidate for a "refactoring" of the "oh, this seems duplicate"-kind :D.